src/expectRevert.js

```diff
@@ -33,23 +33,24 @@ async function checkRevertReasonSupport (provider) {
 
     const nodeInfo = await web3.eth.getNodeInfo();
     const ganacheVersion = /TestRPC\/v([\w.-]+)\/ethereum-js/.exec(nodeInfo);
+    const hardhatVersion = /HardhatNetwork\/([\w.-]+)\/ethereumjs-vm/.exec(nodeInfo);
 
     const warn = function (msg) {
       console.log(`\
 ${colors.white.bgBlack('@openzeppelin/test-helpers')} ${colors.black.bgYellow('WARN')} expectRevert: ${msg}`
       );
     };
 
-    if (ganacheVersion === null) {
+    if (ganacheVersion === null && hardhatVersion === null) {
       warn(`\
 Assertions may yield false negatives!
 
-Revert reason checks are only known to work on Ganache >=2.2.0, and the current node is ${nodeInfo}.
+Revert reason checks are only known to work on Ganache >=2.2.0 and Hardhat >= 2.0.0, and the current node is ${nodeInfo}.
```

Comment:
I think we don't need a version number for Hardhat, as all of them support it

Reply:
Yeah, I initially added it for consistency but it doesn't really make sense. I will remove it.